compiler/src/dotty/tools/dotc/transform/PatternMatcher.scala

```diff
@@ -597,6 +603,16 @@ object PatternMatcher {
             plan
           }
         }
+        override def apply(plan: SeqPlan): Plan = {
```

Comment:
You should update the doc comment of `def inlineVars` saying that it drops unreachable branch as well